### PR TITLE
common arg handling

### PR DESCRIPTION
**Pull request by vpuri3**
move abstol, reltol, maxiters, vebose to cache from wrappers.

ref https://github.com/SciML/LinearSolve.jl/issues/23

### CODE REVIEW

**Comment by vpuri3** [comment]
question: do we want to have common kwarg for gmres_restart across KSP wrappers?

File: src/common.jl [hunk 1]
```diff
@@ -8,6 +8,12 @@ struct LinearCache{TA,Tb,Tu,Tp,Talg,Tc,Tl,Tr}
     isfresh::Bool # false => cacheval is set wrt A, true => update cacheval wrt A
     Pl::Tl        # store final preconditioner here. not being used rn
     Pr::Tr        # wrappers are using preconditioner in cache.alg for now
+
+    # common arguments
+    abstol::Ttol
+    reltol::Ttol
+    maxiters::Ti
```
Comment (ChrisRackauckas) [inline src/common.jl:15]:
```suggestion
    maxiters::Int
```

File: src/common.jl [hunk 2]
```diff
@@ -1,4 +1,4 @@
-struct LinearCache{TA,Tb,Tu,Tp,Talg,Tc,Tl,Tr}
+struct LinearCache{TA,Tb,Tu,Tp,Talg,Tc,Tl,Tr,Ttol,Ti}
```
Comment (ChrisRackauckas) [inline src/common.jl:1]:
```suggestion
struct LinearCache{TA,Tb,Tu,Tp,Talg,Tc,Tl,Tr,Ttol}
```

File: src/common.jl [hunk 3]
```diff
@@ -62,6 +69,11 @@ function SciMLBase.init(prob::LinearProblem, alg::Union{SciMLLinearSolveAlgorith
     A = alias_A ? A : deepcopy(A)
     b = alias_b ? b : deepcopy(b)
 
+
+    abstol   = (abstol   == 0) ? √eps(eltype(A)) : abstol
+    reltol   = (reltol   == 0) ? √eps(eltype(A)) : reltol
```
Comment (ChrisRackauckas) [inline src/common.jl:74]: where does this default come from?

File: src/common.jl [hunk 4]
```diff
@@ -46,6 +52,7 @@ SciMLBase.init(prob::LinearProblem, args...; kwargs...) = SciMLBase.init(prob,no
 
 function SciMLBase.init(prob::LinearProblem, alg::Union{SciMLLinearSolveAlgorithm,Nothing}, args...;
                         alias_A = false, alias_b = false,
+                        abstol=0, reltol=0, maxiters=0, verbose=false,
```
Comment (ChrisRackauckas) [inline src/common.jl:55]: put the defaults up here.

File: src/common.jl (hunk 3 above)
Comment (vpuri3) [reply src/common.jl:74]: https://github.com/JuliaLinearAlgebra/IterativeSolvers.jl/blob/master/src/cg.jl#L122

File: src/common.jl (hunk 4 above)
Comment (vpuri3) [reply src/common.jl]: done

**Comment by vpuri3** [comment]
retriggering CI

**Pull request merged**